src/math/segment.rs

```diff
@@ -5,22 +5,30 @@ use super::Point;
 /// A line segment, defined by its two end points
 #[derive(Clone, Copy, Eq, PartialEq, Hash, Ord, PartialOrd)]
 pub struct Segment<const D: usize> {
-    pub a: Point<D>,
-    pub b: Point<D>,
+    a: Point<D>,
+    b: Point<D>,
 }
 
 impl Segment<2> {
     /// Convert the 2-dimensional segment to a Parry segment
     pub fn to_parry(&self) -> parry2d_f64::shape::Segment {
         [self.a.to_na(), self.b.to_na()].into()
     }
+
+    pub fn points(&self) -> [Point<2>; 2] {
+        [self.a, self.b]
+    }
```

Comment:
I think this method can be unified with the identical one below, by putting it in an `impl<const D: usize> for Segment<D> { ... }` block and returning `[Point<D>; 2]`.

Reply:
I was working on the math code anyway, so I quickly did this in #243.